pkg/addons/addons.go

```diff
@@ -178,15 +178,21 @@ var (
 		return o.GroupVersionKind().GroupKind() == schema.GroupKind{Kind: "Secret"} ||
 			o.GroupVersionKind().GroupKind() == schema.GroupKind{Kind: "ConfigMap"}
 	}
-	nonScFilter = func(o unstructured.Unstructured) bool {
-		return o.GroupVersionKind().Group != "servicecatalog.k8s.io"
+	nonCRDFilter = func(o unstructured.Unstructured) bool {
+		return o.GroupVersionKind().Group != "servicecatalog.k8s.io" &&
+			o.GroupVersionKind().Group != "monitoring.coreos.com"
 	}
+
 	scFilter = func(o unstructured.Unstructured) bool {
 		return o.GroupVersionKind().Group == "servicecatalog.k8s.io"
 	}
 	crdFilter = func(o unstructured.Unstructured) bool {
 		return o.GroupVersionKind().GroupKind() == schema.GroupKind{Group: "apiextensions.k8s.io", Kind: "CustomResourceDefinition"}
 	}
+	// targeted filter is used to target specific CRD's, which are managed not by sync pod
+	targetedCrdFilter = func(o unstructured.Unstructured) bool {
```

Comment:
I suspect we will have more of these, with the same condition. So instead of duplicating and adding more filers, more `updateDynamicClient` and more `CRDReady` statements to follow I was thinking to keep this generic and add all under one filter?

Reply:
the filter thing is exploding, but I don't think your way is the way to solve the problem.  Keeping these separate at least makes dependencies clearer - also helps my work to make the sync pod report ready when it is ready (excluding waiting for the monitoring stack at cluster startup)